trin-history/src/content_key.rs

```diff
@@ -0,0 +1,148 @@
+use sha2::{Digest, Sha256};
+use ssz::{self, Decode, Encode};
+use ssz_derive::{Decode, Encode};
+use trin_core::portalnet::types::content_key::OverlayContentKey;
+
+/// A content key in the history overlay network.
+#[derive(Clone, Debug, Decode, Encode)]
+#[ssz(enum_behaviour = "union")]
+pub enum HistoryContentKey {
+    /// A block header.
+    BlockHeader(BlockHeader),
+    /// A block body.
+    BlockBody(BlockBody),
+    /// The transaction receipts for a block.
+    BlockReceipts(BlockReceipts),
+}
+
+/// A key for a block header.
+#[derive(Clone, Debug, Decode, Encode)]
+pub struct BlockHeader {
+    /// Chain identifier.
+    chain_id: u16,
+    /// Hash of the block.
+    block_hash: [u8; 32],
+}
+
+/// A key for a block body.
+#[derive(Clone, Debug, Decode, Encode)]
+pub struct BlockBody {
+    /// Chain identifier.
+    chain_id: u16,
+    /// Hash of the block.
+    block_hash: [u8; 32],
+}
+
+/// A key for the transaction receipts for a block.
+#[derive(Clone, Debug, Decode, Encode)]
+pub struct BlockReceipts {
+    /// Chain identifier.
+    chain_id: u16,
+    /// Hash of the block.
+    block_hash: [u8; 32],
+}
+
+#[allow(clippy::from_over_into)]
+impl Into<Vec<u8>> for HistoryContentKey {
+    fn into(self) -> Vec<u8> {
+        self.as_ssz_bytes()
+    }
+}
+
+impl TryFrom<Vec<u8>> for HistoryContentKey {
+    type Error = &'static str;
```

Comment:
I feel like we've been trying to move towards custom error types in this codebase. Using `String` above for `MockContentKey` in the tests seems perfectly reasonable, but is there a reason you didn't go for a custom error here?

Reply:
So in the trait definition of `OverlayContentKey`, I didn't specify a bound on the `TryFrom::Error` associated type for flexibility, but looking at things again, that's not so useful, and we might want some kind of enum like `ContentKeyDecodeError` to provide more useful information. That, or we should at least pass through the `ssz::DecodeError`.